src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -1105,6 +1115,27 @@ private static FunctionResult LoadContentResultBuilder(ObjectParser objectParser
             }
             return new(ErrorType.Create(errorDiagnostic));
         }
+
+          private static FunctionResult ReadEnvironmentVariableResultBuilder (IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, FunctionCallSyntaxBase functionCall, ImmutableArray<TypeSymbol> argumentTypes)
+         {
+            var arguments = functionCall.Arguments.ToImmutableArray();
+           
+            if (argumentTypes.Length != 1 || argumentTypes[0] is not StringLiteralType stringLiteral)
+            {
+                return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[0]).CompileTimeConstantRequired()));
+            }
+            var envVariableName = stringLiteral.RawStringValue;
+            var envVariableValue = Environment.GetEnvironmentVariable(envVariableName);
+            
+            //error to fail the build-param with clear message of the missing env var name
+            if (envVariableValue == null)
+            {
+                return new(ErrorType.Create(DiagnosticBuilder.ForPosition(arguments[0]).FailedToEvaluateParameter(envVariableName, "Environment variable does not exist")));
```

Comment:
Original plan(1) was using null for missing environment variables. Is there a reason to emit en error?

Throwing an error blocks setting default values with ternary operator or null coalescing.


(1)

Reply:
Looking at the possible TypeSymbols available at of today Bicep does not provide a nullable TypeSymbol return.

Reply:
It could be a good idea to make a poll in community call what users wants

* emit error on missing environment variables + setting default parameter value on bicep file
* or use usee ternary operator/null coalescing and set default value on .bicepparam file

Luckily there is still some time to change this afterwards before breaking change enforcements.